### DISCUSSION TITLE
Error when using Langchain js HNSWLib/MemoryVectorStor

### DISCUSSION THREAD

**Discussion by masasso** [microsoft/vscode-discussions#587] [Extension Development QnA] [answered]
I am trying to use HNSWLib for the code below.
`
import { HNSWLib } from "langchain/vectorstores/hnswlib";
import { OpenAIEmbeddings } from "langchain/embeddings/openai";
console.log("Start HNSWLib vectorstore");

    const vectorStore = await HNSWLib.fromTexts(
        ["Hello world", "Bye bye", "hello nice world"],
        [{ id: 2 }, { id: 1 }, { id: 3 }],
        new OpenAIEmbeddings()
    );
    console.log("HNSWLib similaritySearch");
    const resultOne = await vectorStore.similaritySearch("hello world", 1);
    console.log("My result");
    console.log(resultOne);`

In my debug window I get this error:
`Activating extension 'Extension.Extension' failed: Headers is not defined.`

This is the error I am getting after setting  `process.env['NODE_OPTIONS'] = "--experimental-fetch"`:

```
rejected promise not handled within 1 second: ReferenceError: Headers is not defined extensionHostProcess.js:102 stack trace: ReferenceError: Headers is not defined at createRequest (/home/matheus/<package>/dist/extension.js:29734:21) at fetchAdapter (/home/matheus/<package>/dist/extension.js:29655:21) at dispatchRequest (/home/matheus/<package>/dist/extension.js:27012:10) at Axios.request (/home/matheus/Projects/<package>/dist/extension.js:26769:15) at Function.wrap [as request] (/home/matheus/<package>n/dist/extension.js:26652:15) at /home/matheus/Projects/<package>/dist/extension.js:28819:22 at /home/matheus/Projects/<package>/dist/extension.js:25955:123 at async RetryOperation._fn (/home/matheus/Projects/<package>/dist/extension.js:30236:12)
```

I did the same test for Langchain MemoryVectorStore and the same error is happening.

The point is. I did a simple node js app and it worked, so **this error is happening only in vscode extension**

Even thought in my package.json I configure node to be in v.18, inside vscode extension debug window it behaviors as if it was in v16. I don't know exactly if that happens because those libraries are built in C++ or if in vscode extension debug window it is truly using v16.

I am suspecting that inside **vscode extension** debug environment it has node 16 because I am getting the exact behavior that the package describes for v16:     https://js.langchain.com/docs/getting-started/install

**Comment by isidorn** [comment, marked as answer]
Yes, it might be that you are getting in the VS Code Extension Debug Env node 16. Depends on the electron we are using.
Though I think this should no longer repro since we have updated our Electron.
